pom.xml

```diff
@@ -166,20 +167,43 @@
                     </fileset>
                   </path>
                   <sequential>
+
+                    <!-- unset variable to ensure it is properly updated in next property and dirname tasks -->
+                    <var name="path_to_copy_to" unset="true"/>
+                    <var name="file_path" unset="true"/>
+
+
+                    <property name="path_to_copy_to" value=""/>
+                    <dirname property="file_path" file="@{asciidoc-file}"/>
+
+                    <!-- Remove the path from the root directory until current sourceDirectory to obtain a relative path -->
+                    <script language="javascript">
+                      // overwrite existing property.
+                      project.setProperty(
+                        'path_to_copy_to',
+                        project.getProperty('file_path').replace(
+                          project.getProperty('basedir'),
+                          ''
+                        ).replace(/\\/g, '/').replace('/', ''));
+                    </script>
+
                     <!-- unset variable to ensure it is properly updated in next basename task -->
                     <var name="filename" unset="true"/>
                     <!--<basename property="filename" file="@{asciidoc-file}" suffix=".asciidoc"/>-->
                     <basename property="filename" file="@{asciidoc-file}"/>
-                    <copy file="@{asciidoc-file}" todir="${docgen.asciidoc.target}" encoding="${project.build.sourceEncoding}" outputencoding="${project.reporting.outputEncoding}" verbose="true">
+                    <copy file="@{asciidoc-file}" todir="${docgen.asciidoc.target}${path_to_copy_to}" encoding="${project.build.sourceEncoding}" outputencoding="${project.reporting.outputEncoding}" verbose="true">
                       <filterchain>
                         <tokenfilter>
                           <linetokenizer/>
                           <!-- make anchors unique by prefixing with filename -->
                           <replaceregex pattern="\[\[" replace="[[${filename}_" flags="g"/>
                           <!-- automatically generate anchors for sections -->
                           <replaceregex pattern="^(==+) (.*)" replace="[[${filename}_\2]]${line.separator}\1 \2" flags="g"/>
+
                           <!-- fix includes to contain asciidoc suffix (what is omitted on github wiki) -->
-                          <replaceregex pattern="include\:\:([^.]*)\[" replace="include::\1.asciidoc[" flags="g"/>
+                          <replaceregex pattern="include\:\:(.*)\[" replace="include::\1.asciidoc[" flags="g"/>
```

Comment:
couldn't this line be removed in favor of the new more generic replaceregex?

Reply:
This line can't be removed. See the following example:

1) Posible initial situations:

include::something
include::something.asc
include::something.adoc
include::something.asciidoc


2) after "&lt;replaceregex pattern="include\:\:(.*)\[" replace="include::\1.asciidoc[" flags="g"/&gt;":

include::something.asciidoc
include::something.asc.asciidoc
include::something.adoc.asciidoc
include::something.asciidoc.asciidoc


3) after "&lt;replaceregex pattern="include\:\:(.*)((\.adoc)|(\.asc)|(\.asciidoc))(\.asciidoc)\[" replace="include::\1\2[" flags="g"/&gt;"


include::something.asciidoc
include::something.asc
include::something.adoc
include::something.asciidoc


Line 2) is neccesary because it adds the necessary .asciidoc extension if it is not added. If the included file already has a extension, it still adds .asciidoc but will be remove by 3)

If the modified _include_ has a leftover .asciidoc extension (.asc.asciidoc, .adoc.asciidoc, .asciidoc.asciidoc), line 3) removes it maintaining the original .asc, .adoc and .asciidoc in files that had no extension initially


It had to be done this way because with the original aproach, "pattern="include\:\:([^.]*)\["" was not adding .asciidoc if there was a dot (.) in the include statement:

Original:
include::my.wiki.folder/something ---> result --> include::my.wiki.folder/something (no .asciidoc)

Now:
include::my.wiki.folder/something ---> result --> include::my.wiki.folder/something.asciidoc

Reply:
ok got it, thanks for the explanation!